File: backend/models/Category.js
const mongoose = require('mongoose');

const categorySchema = new mongoose.Schema({
  name: {
    type: String,
    required: [true, 'Category name is required'],
    trim: true,
    unique: true
  },
  slug: {
    type: String,
    required: [true, 'Slug is required'],
    unique: true,
    lowercase: true,
    trim: true
  },
  icon: {
    type: String,
    default: ''
  },
  image: {
    type: String,
    default: ''
  },
  description: {
    type: String,
    trim: true
  },
  visibility: {
    type: String,
    enum: ['visible', 'hidden'],
    default: 'visible'
  }
}, {
  timestamps: true
});

// Generate slug from name before saving
categorySchema.pre('save', function(next) {
  if (this.isModified('name') && !this.slug) {
    this.slug = this.name.toLowerCase().replace(/[^a-z0-9]+/g, '-').replace(/(^-|-$)/g, '');
  }
  next();
});

module.exports = mongoose.model('Category', categorySchema);

File: lib/models/category.dart
class Category {
  final String id;
  final String name;
  final String slug;
  final String? description;
  final String? icon;
  final bool isVisible;

  Category({
    required this.id,
    required this.name,
    required this.slug,
    this.description,
    this.icon,
    this.isVisible = true,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['_id'] ?? '',
      name: json['name'] ?? 'General',
      slug: json['slug'] ?? (json['name'] ?? '').toString().toLowerCase().replaceAll(' ', '-'),
      description: json['description'],
      icon: json['icon'],
      isVisible: json['visibility'] == 'visible' || json['visibility'] == true,
    );
  }
}
